runtime/src/main/java/org/corfudb/protocols/logprotocol/SMREntry.java

```diff
@@ -120,27 +124,33 @@ void deserializeBuffer(ByteBuf b, CorfuRuntime rt) {
         SMRMethod = new String(methodBytes);
         byte serializerId = b.readByte();
         byte numArguments = b.readByte();
-        Object[] arguments = new Object[numArguments];
+        List<Object> arguments = new ArrayList<>();
 
         if (!opaque) {
             serializerType = Serializers.getSerializer(serializerId);
         } else {
             this.serializerId = serializerId;
         }
 
+        List<Pair<ByteBuf, Integer>> objBufPairs= new ArrayList<>();
         for (byte arg = 0; arg < numArguments; arg++) {
             int len = b.readInt();
             ByteBuf objBuf = b.slice(b.readerIndex(), len);
+            objBufPairs.add(new ImmutablePair<>(objBuf, len));
```

Comment:
Can you explain why `len` needs to be stored? Can it be inferred from `readableBytes()`?

Reply:
I didn't look into the ByteBuf class and simply wrap them to a Pair. Thanks for pointing it out, just wrote a test to verify it and will remove it.

```
@Test
    public void testByteBuffer() {
        ByteBuf byteBuf = Unpooled.buffer();
        byte[] bytes = "testByteBuffer".getBytes();
        Serializers.CORFU.serialize(bytes, byteBuf);
        int len = 5;
        ByteBuf slice = byteBuf.slice(byteBuf.readerIndex(), len);
        assertThat(slice.readableBytes()).isEqualTo(len);
    }
```